### PR TITLE
kubevirt-console-plugin: Use two replicas in case of highly available cluster

### PR DESCRIPTION
**Pull request by orenc1**
In case the cluster has an HighlyAvailable infrastructure, the kubevirt-console-plugin and kubevirt-console-proxy should be highly available as well, in order to avoid losing the Virtualization plugin when the hosting node is down. By running these pods with two replicas and PodAntiAffinity rule, we can ensure the plugin will remain available when a node is down.

**What this PR does / why we need it**:

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly

**Jira Ticket**:

```jira-ticket
https://issues.redhat.com/browse/CNV-45427
```

**Release note**:

```release-note
Adjust kubevirt-console-plugin for high availability
```

### CODE REVIEW

**Review by nunnatsa — COMMENTED** [review]
Looks good!

Please just add some tests to cover this change.

**Comment by orenc1** [comment]
> Looks good!
> 
> Please just add some tests to cover this change.

Done, unit test cases have been added to cover the automatic PodAntiAffinity and replicas configuration.

**Review by nunnatsa — COMMENTED** [review]
Added several inline comments. In addition, WDYT about adding two very short e2e tests, one with `Label(tests.SingleNodeLabel)` and the other with  `Label(tests.HighlyAvailableClusterLabel)`, to check how many pods are running and/or the deployment's `replicas` field?

File: controllers/operands/kubevirtConsolePlugin_test.go [hunk 1]
```diff
@@ -686,6 +686,84 @@ var _ = Describe("Kubevirt Console Plugin", func() {
...
+				Entry("plugin deployment", hcoutil.AppComponentUIPlugin, NewKvUIPluginDeployment, newKvUIPluginDeploymentHandler),
+				Entry("proxy deployment", hcoutil.AppComponentUIProxy, NewKvUIProxyDeployment, newKvUIProxyDeploymentHandler),
+			)
+
+			DescribeTable("use one replica on SNO", func(appComponent hcoutil.AppComponent,
+				deploymentManifestor func(converged *hcov1beta1.HyperConverged) *appsv1.Deployment, handlerFunc GetHandler) {
+
+				hcoutil.GetClusterInfo = func() hcoutil.ClusterInfo {
+					return &commontestutils.ClusterInfoSNOMock{}
+				}
```
Comment (nunnatsa) [inline controllers/operands/kubevirtConsolePlugin_test.go:737]: Please restore `hcoutil.GetClusterInfo` - as above

File: controllers/operands/kubevirtConsolePlugin.go [hunk 2]
```diff
@@ -195,12 +204,37 @@ func getKvUIDeployment(hc *hcov1beta1.HyperConverged, deploymentName string, ima
...
+						},
+						TopologyKey: corev1.LabelHostname,
+					},
+				},
+			},
+		}
+	}
+
+	return nil
+}
```
Comment (nunnatsa) [inline controllers/operands/kubevirtConsolePlugin.go:236]: not sure I want this function here. we only use it in the new tests. let's move it there

**Review by nunnatsa — COMMENTED** [review]
Two more inline comments

File: controllers/operands/kubevirtConsolePlugin.go (hunk 2 above)
Comment (orenc1) [reply controllers/operands/kubevirtConsolePlugin.go:236]: where should i move it to? `testUtils.go` ?

Comment (nunnatsa) [reply controllers/operands/kubevirtConsolePlugin.go:236]: to the only file that uses it

Comment (orenc1) [reply controllers/operands/kubevirtConsolePlugin.go:236]: could you please suggest where this function should be move into?
note that it isn't used by tests only.

Comment (orenc1) [reply controllers/operands/kubevirtConsolePlugin.go:236]: >to the only file that uses it

the only usage of that function is in the same file - `kubevirtConsolePlugin.go`

Comment (nunnatsa) [reply controllers/operands/kubevirtConsolePlugin.go:236]: sorry. misread

File: controllers/commontestutils/testUtils.go [hunk 3]
```diff
@@ -141,6 +142,27 @@ func NewOtherNodePlacement() *sdkapi.NodePlacement {
...
+							},
+						},
+					},
+					TopologyKey: corev1.LabelHostname,
+				},
+			},
+		},
+	}
+}
+
```
Comment (nunnatsa) [inline controllers/commontestutils/testUtils.go:165]: oh, this is the function I really want to comment about. sorry, these two functions share similar visual (:

So I would move **this one** to the file where it actually used, and make it also not exported.

Comment (orenc1) [reply controllers/commontestutils/testUtils.go]: makes sense.
the function has been moved :+1:

**Pull request merged**